mpas_analysis/ocean/meridional_overturning_circulation.py

```diff
@@ -483,18 +463,18 @@ def write_file(dsMOCTimeSeries):
             first = False
         date = days_to_datetime(ds.Time[tIndex], calendar=calendar)
         print '     date: {:04d}-{:02d}'.format(date.year, date.month)
-        horizontalVel = ds.avgNormalVelocity[tIndex, :, :].values
-        verticalVel = ds.avgVertVelocityTop[tIndex, :, :].values
-        velArea = verticalVel * areaCell[:, np.newaxis]
-        transportZ = _compute_transport(maxEdgesInTransect,
-                                        transectEdgeGlobalIDs,
-                                        transectEdgeMaskSigns,
+        # Cache results if their size is reasonable
+        load = partial(preload, maxgb=config.getfloat('input',
+                                                      'reasonableArraySizeGB'))
+        horizontalVel = load(ds.avgNormalVelocity.isel(Time=tIndex))
+        velArea = load(ds.avgVertVelocityTop.isel(Time=tIndex) * areaCell)
```

Comment:
The reason I did it this way is to parallel the xarray `load` method.  Also, it seems like `reasonableArraySizeGB` isn't really something we need to change per call to preload.  But, `partial` is confusing for first-time users.  I also thought about making `reasonableArraySizeGB` a global but decided against it for obvious reasons.

Do you think this needs changed?